sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/MethodIdProvider_Create.cs

```diff
@@ -0,0 +1,192 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+using csharp::SonarAnalyzer.Security.Ucfg;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarAnalyzer.UnitTest.Security.Ucfg
+{
+    [TestClass]
+    public class MethodIdProvider_Create
+    {
+        [TestMethod]
+        public void MethodIdProvider_Create_Methods()
+        {
+            const string code = @"
+using System;
+namespace Namespace
+{
+    public class Class1
+    {
+        public void PrimitiveTypes1(string s) { }
+        public void PrimitiveTypes2(String s) { }
+        public void PrimitiveTypes3(System.String s) { }
+
+        public void ArrayTypes(string[] s) { } // TODO: improve array type signature
```

Comment:
What does improve in that TODO mean ? shouldn't  we have a ticket for this improvement ? (and so it could be put in the TODO comment).

Reply:
Currently we generate `__unknown` as TypeId (see the assert below, you could look for "ArrayTypes") but while I am thinking about it, I am more convinced that we should resolve all kinds of types from the beginning: https://jira.sonarsource.com/browse/SONARSEC-49